### PR TITLE
Magic Links: Wires new Endpoints

### PR DESCRIPTION
**Pull request by jleandroperez**
### ⚠️ Important
- [x] Make sure the new endpoints are deployed before merging
- [x] Remove the staging URL!

### Fix
In this PR we're wiring the new code-based endpoints.

Ref. #1187

### Test
1. Logout from Simplenote macOS
2. Click on the `Log In` button
3. Enter your email
4. Click on `Instantly Log In with Email`

- [ ] Verify that shortly after you get the Auth Email
- [ ] Verify that clicking on `Log Into Simplenote` button causes the mac app to Log In
- [ ] Verify the Unit Test pass

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Simplenote/SimplenoteConstants.swift [hunk 1]
```diff
@@ -31,8 +31,8 @@ class SimplenoteConstants: NSObject {
 
     /// URL(s)
     ///
-    static let loginRequestURL              = googleAppEngineBaseURL.appendingPathComponent("/account/request-login")
-    static let loginCompletionURL           = googleAppEngineBaseURL.appendingPathComponent("/account/complete-login")
+    static let loginRequestURL              = "https://magic-code-dot-simple-note-hrd.appspot.com/account/request-login" //googleAppEngineBaseURL.appendingPathComponent("/account/request-login")
+    static let loginCompletionURL           = "https://magic-code-dot-simple-note-hrd.appspot.com/account/complete-login" //googleAppEngineBaseURL.appendingPathComponent("/account/complete-login")
```
Comment (jleandroperez) [inline Simplenote/SimplenoteConstants.swift]: 🛑 Important!! This will be reverted before merging!!

**Review by charliescheer — CHANGES_REQUESTED** [review]
Works as intended, but the unit tests are failing locally and on the CI

**Comment by jleandroperez** [comment]
Tests fixed, ready for another look. 

Thank you Charlie!!

**Review by charliescheer — APPROVED** [review]
Tests succeed now, thanks!

**Comment by jleandroperez** [comment]
Thank youuu!!

**Pull request merged**